packages/react-draggable-dialog/src/components/DraggableDialog/useDraggableDialog.ts

```diff
@@ -0,0 +1,177 @@
+import * as React from 'react';
+import { useFluent, useEventCallback } from '@fluentui/react-components';
+
+function roundByDPR(value: number, document?: Document) {
+  const win = document?.defaultView;
+
+  if (!win) {
+    return value;
+  }
+
+  const dpr = win.devicePixelRatio || 1;
+
+  return Math.round(value * dpr) / dpr;
+}
+
+type DraggableMarginAxis = {
+  mainAxis?: number;
+  crossAxis?: number;
+};
+type DraggableMarginViewport = {
+  top?: number;
+  end?: number;
+  bottom?: number;
+  start?: number;
+};
+type DraggableDialogOptions = {
+  keepInViewport?: boolean;
+  margin?: number | DraggableMarginAxis | DraggableMarginViewport;
+};
+
+const getDraggableMargin = (
+  margin: DraggableDialogOptions['margin']
+): Required<DraggableMarginViewport> => {
+  const defaultMargin = {
+    top: 0,
+    end: 0,
+    bottom: 0,
+    start: 0,
+  };
+
+  if (!margin) {
+    return defaultMargin;
+  }
+
+  if (typeof margin === 'number') {
+    return {
+      top: margin,
+      end: margin,
+      bottom: margin,
+      start: margin,
+    };
+  }
+
+  if ('mainAxis' in margin || 'crossAxis' in margin) {
+    const x = margin.mainAxis || 0;
+    const y = margin.crossAxis || 0;
+
+    return {
+      top: y,
+      end: x,
+      bottom: y,
+      start: x,
+    };
+  }
+
+  return {
+    ...defaultMargin,
+    ...margin,
+  };
+};
+
+export const useDraggableDialog = (_opts: DraggableDialogOptions) => {
+  const { targetDocument } = useFluent();
+  const { margin, keepInViewport } = {
+    keepInViewport: true,
+    ..._opts,
+    margin: getDraggableMargin(_opts.margin),
+  };
+  const [dragging, setDragging] = React.useState(false);
+  const ref = React.useRef<HTMLDivElement>(null);
+  const currentPosition = React.useRef({ x: 0, y: 0 });
+
+  const getInBoundsPosition = useEventCallback(
+    (event: React.MouseEvent<HTMLDivElement>) => {
+      const {
+        currentTarget: { offsetLeft, offsetTop },
+      } = event;
+
+      let { x, y } = {
+        x: event.clientX - currentPosition.current.x,
+        y: event.clientY - currentPosition.current.y,
+      };
+
+      if (!keepInViewport) {
+        return { x, y };
+      }
+
+      if (x - margin.start + offsetLeft < 0) {
```

Comment:
im wondering if all those checks are necessary to evaluate on every function call ?

Reply:
This file is now removed.